File: doc/src/scripts/RemoteTechExample.cpp
#include <iostream>
#include <krpc.hpp>
#include <krpc/services/space_center.hpp>
#include <krpc/services/remote_tech.hpp>

int main() {
  krpc::Client conn = krpc::connect("RemoteTech Example");
  krpc::services::SpaceCenter space_center(&conn);
  krpc::services::RemoteTech remote_tech(&conn);
  auto vessel = space_center.active_vessel();

  // Set a dish target
  auto part = vessel.parts().with_title("Reflectron KR-7").front();
  auto antenna = remote_tech.antenna(part);
  antenna.set_target_body(space_center.bodies()["Jool"]);

  // Get info about the vessels communications
  auto comms = remote_tech.comms(vessel);
  std::cout << "Signal delay = " << comms.signal_delay() << std::endl;
}

File: doc/src/scripts/RemoteTechExample.py
import krpc
conn = krpc.connect(name='RemoteTech Example')
vessel = conn.space_center.active_vessel

# Set a dish target
part = vessel.parts.with_title('Reflectron KR-7')[0]
antenna = conn.remote_tech.antenna(part)
antenna.target_body = conn.space_center.bodies['Jool']

# Get info about the vessels communications
comms = conn.remote_tech.comms(vessel)
print 'Signal delay =', comms.signal_delay
